### ISSUE TITLE
keyd-application-mapper support for Gnome 42

### ISSUE THREAD

**Issue by canadaduane** [rvaiya/keyd#223]
I just switched to Wayland and tried installing keyd-application-mapper.

The first thing I noticed is that the Gnome extension is hard-coded at compatibility with Gnome 41. I modified the metadata.json to say it's compatible with 42.

Next, I noticed that the names of applications are different when coming from the Gnome extension vs. the X11 variant. For example:

`gnome-terminal` (x11) vs `gnome-terminal-server` (gnome/wayland)

Are there any knobs or levers we can pull to make these the same, regardless of DE, or should I just plan on testing each app in both and seeing what is produced? (e.g. I can use `[gnome-terminal*]` instead of `[gnome-terminal]` in the keyd-application-mapper's `app.conf` file to cover both cases for the Gnome Terminal; however, this technique cannot be applied universally).

**Comment by rvaiya** [comment]
> The first thing I noticed is that the Gnome extension is hard-coded at compatibility with Gnome 41. I modified the metadata.json to say it's compatible with 42.

Thanks for reporting this. I will release a fix for this shortly (I have some other changes sitting in my tree).

> gnome-terminal (x11) vs gnome-terminal-server (gnome/wayland)

From what I can see gnome-terminal uses a different class depending on whether [or not](https://gitlab.gnome.org/GNOME/gnome-terminal/-/blob/master/src/terminal.cc#L555) it is run as a [server](https://gitlab.gnome.org/GNOME/gnome-terminal/-/blob/master/src/server.cc#L151), which I'm assuming is indirectly affected by whether or not you are running it in wayland. 

> Are there any knobs or levers we can pull to make these the same, regardless of DE

My understanding is that mutter derives the class either from the wayland app id or from the WM_CLASS property (in the case of X clients). Most applications don't have native wayland support which means that they will run under XWayland, which in turn uses the WM_CLASS property as the app id. For those that do have native support it would seem unusual for the app id not to reflect the old X class property, though there is nothing which forces them to be identical.

TLDR: In most cases the class should be the same, but the only way to be sure is to test :P.

**Comment by canadaduane** [comment]
Ahh, that makes sense! Thanks for identifying the root cause re: difference in gnome-terminal names. I suspect this won't be as large a problem as I originally thought. Wildcards are working fine for now.

And I see you updated the compatibility metadata for Gnome 42. I think we can close this!
